iceoryx_examples/automotive_soa/include/minimal_proxy.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_MINIMAL_PROXY_HPP
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_MINIMAL_PROXY_HPP
+
+#include "topic.hpp"
+
+#include "owl/kom/event_subscriber.hpp"
+#include "owl/kom/field_subscriber.hpp"
+#include "owl/kom/method_client.hpp"
+#include "owl/runtime.hpp"
+#include "owl/types.hpp"
+
+/// @note Once a handler has been set with 'EnableFindServiceCallback', calling 'FindService' is not thread-safe!
+class MinimalProxy
+{
+  public:
+    static constexpr char m_serviceIdentifier[] = "MinimalSkeleton";
+
+    MinimalProxy(owl::kom::ProxyHandleType& handle) noexcept;
+    ~MinimalProxy() noexcept = default;
+
+    MinimalProxy(const MinimalProxy&) = delete;
+    MinimalProxy(MinimalProxy&&) = delete;
+    MinimalProxy& operator=(const MinimalProxy&) = delete;
+    MinimalProxy& operator=(MinimalProxy&&) = delete;
+
+    static owl::kom::FindServiceCallbackHandle
+    EnableFindServiceCallback(owl::kom::FindServiceCallback<owl::kom::ProxyHandleType> handler,
+                              owl::kom::InstanceIdentifier& instanceIdentifier) noexcept;
+
+    static void DisableFindServiceCallback(owl::kom::FindServiceCallbackHandle handle) noexcept;
+
+    static owl::kom::ServiceHandleContainer<owl::kom::ProxyHandleType>
+    FindService(owl::kom::InstanceIdentifier& instanceIdentifier) noexcept;
+
+    const owl::kom::InstanceIdentifier m_instanceIdentifier;
+    owl::kom::EventSubscriber<TimestampTopic1Byte> m_event{m_serviceIdentifier, m_instanceIdentifier, "Event"};
```

Comment:
Could we introduce some variables like `constexpr const char EVENT_IDENTIFIER[], FIELD_IDENTIFIER[], METHOD_IDENTIFIER[]` in `owl/globals.hpp` and store the values there.
Otherwise a type in a magic value like this could lead to unnecessary bugs and it looks cleaner.

Reply:
See my answer in the [previous comment](https://github.com/eclipse-iceoryx/iceoryx/pull/1355#discussion_r886603361). Proxy and skeleton code typically is never written by the users.